### ISSUE TITLE
Logging in should also trigger a refresh

### ISSUE THREAD

**Issue by artlowel** [DSpace/dspace-angular#712]
When logging in, we should also do a refresh, the same way we do when logging out. Because rest requests that were made while anonymous may have different results when you're logged in.

E.g. the bitstream of [this item](https://dspace7-demo.atmire.com/items/6160810f-1e53-40db-81ef-f6621a727398) is private, so if you go to this page anonymously you don't see a bitstream.

If you log in using the menu in the header you still don't see the bitstream. Only after you refresh do you see it.

While this is one small example, with the support for features and authorizations in the UI a lot more of these will start to surface. The most foolproof way of dealing with this is refreshing whenever the active user changes in my opinion.

**Note:** I just created another issue (#711) involving redirects after logging in. Perhaps fixing this issue can also solve that one.
